tests/unit/pages/tutor-profile/TutorProfile.spec.jsx

```diff
@@ -0,0 +1,85 @@
+import { vi } from 'vitest'
+import { renderWithProviders } from '~tests/test-utils'
+import { screen } from '@testing-library/react'
+import TutorProfile from '~/pages/tutor-profile/TutorProfile.jsx'
+import useAxios from '~/hooks/use-axios'
+
+const route = '/tutor/my-profile'
+const appMain = {
+  userRole: 'tutor',
+  _id: '648850c4fdc2d1a130c24aea'
+}
+
+const mockData = {
+  firstName: 'Іван',
+  lastName: 'Мавдрик',
+  mainSubjects: {
+    student: [],
+    tutor: [
+      {
+        _id: '648850c4fdc2d1a130c24aea',
+        name: 'Guitar',
+        category: '64884f21fdc2d1a130c24ac0',
+        totalOffers: {
+          student: 4,
+          tutor: 12
+        }
+      },
+      {
+        _id: '64885108fdc2d1a130c24af9',
+        name: 'Cybersecurity',
+        category: '64884f33fdc2d1a130c24ac2',
+        totalOffers: {
+          student: 3,
+          tutor: 20
+        }
+      }
+    ]
+  },
+  totalReviews: {
+    student: 0,
+    tutor: 0
+  },
+  averageRating: {
+    student: 0,
+    tutor: 0
+  }
+}
+
+vi.mock('~/hooks/use-axios')
+
+describe('TutorProfile', () => {
+  it('loading loader', () => {
+    const fakeData = {
+      loading: true,
+      response: mockData
+    }
+
+    useAxios.mockImplementation(() => fakeData)
+    renderWithProviders(<TutorProfile />, {
+      initialEntries: route,
+      preloadedState: { appMain }
+    })
+
+    const loader = screen.getByTestId('loader')
+
+    expect(loader).toBeInTheDocument()
+  })
+
+  it('should find rendering name', () => {
+    const fakeData = {
```

Comment:
I think you can create a function getFakeData which accepts a boolean value for loading and returns this object

Reply:
done